### ISSUE TITLE
Considerations regarding output buffers

### ISSUE THREAD

**Issue by mara004** [unoconv/unoserver#38]
While studying [the converter code](https://github.com/unoconv/unoserver/blob/03a199ee51935bcb8104842027946f737e8e626c/src/unoserver/converter.py), I wanted to share some thoughts regarding output buffers:
Currently, if one does not want to write to a file, it's possible to set `outpath` to `None`. In this case, the data will be written into a new, internally created `io.BytesIO` object, and then the value of the buffer is returned as `bytes`.

However, that means the entire data will be in memory at the same time, which increases resource usage.
Looking at the uno outputstream interface, it seems like the output is actually provided incrementally, though (presuming that `writeBytes()` is called multiple times with smaller parts rather than one large sequence):
```python
class OutputStream(unohelper.Base, XOutputStream):
    def __init__(self):
        self.buffer = io.BytesIO()

    def closeOutput(self):
        pass

    def writeBytes(self, seq):
        self.buffer.write(seq.value)
```

If unoserver would accept a caller-provided output buffer to write into (e. g. a file handle acquired by `open(..., "wb")`, or `sys.stdout`), the data wouldn't necessarily have to be in memory at once.

Considering a possible backwards-compatible implementation, `outpath` could just be adapted to accept a byte buffer (i. e. anything that implements `write()`, `read()`, and `seek()`), and an initialisation parameter could be added to `OutputStream` to take it over.

I'm not certain how useful this would be, given that uno can already write to files on its own, and if you intend to post-process the output, it probably needs to be in memory as a whole anyway. Nevertheless, it seems a bit more elegant, and might provide callers with a higher level of control over file creation. Would you be interested in a Pull Request?

**Comment by regebro** [comment]
Yes, PDF's will be sent one page at a time, for example, so that could be a useful Pull Request.
